### PR TITLE
Feature - Django Storages Backends

### PR DESCRIPTION
**Pull request by CamiloValderruten**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -39,9 +39,9 @@
     'psycopg2',
 
     # storage
-    'django-storages<1.9',
-    'boto>=2.40.0',
-    'djeese-fs',
+    'django-storage-url',
+    'boto3',
```
Comment (CamiloValderruten) [inline setup.py]: Move to django storages url

File: aldryn_django/storage.py [hunk 2]
```diff
@@ -1,199 +1,47 @@
...
-        # string if it does not end with a '/'
-        config['MEDIA_URL'] = '{}/'.format(config['MEDIA_URL'])
-    return config
+def is_default_storage_on_other_domain():
+    media_host = furl.furl(DefaultStorage().base_url).host
+    return media_host not in settings.ALLOWED_HOSTS if media_host else False
 
 
-class GZippedStaticFilesMixin(object):
+class GZippedStaticFilesMixin:
```
Comment (CamiloValderruten) [inline aldryn_django/storage.py]: Leave object for python2

File: aldryn_config.py [hunk 3]
```diff
@@ -370,18 +373,60 @@ def sentry_settings(self, settings, env):
...
+        settings['MEDIA_URL_IS_ON_OTHER_DOMAIN'] = env('MEDIA_URL_IS_ON_OTHER_DOMAIN', None)
+        if not settings['MEDIA_URL_IS_ON_OTHER_DOMAIN']:
+            settings['MEDIA_URL_IS_ON_OTHER_DOMAIN'] = lazy_setting(
+                'MEDIA_URL_IS_ON_OTHER_DOMAIN',
+                is_default_storage_on_other_domain,
+                bool,
+            )
+
+        # TODO: Not sure if this line is still required.
+        # settings['MEDIA_ROOT'] = env('MEDIA_ROOT', os.path.join(settings['DATA_ROOT'], 'media'))
```
Comment (CamiloValderruten) [inline aldryn_config.py]: Yes, needed

Comment (CamiloValderruten) [reply aldryn_config.py]: Local dev

**Comment by FinalAngel** [comment]
@CamiloValderruten seems like your branch is a bit out of date with "support/2.2.x" because I still see the env SITE_ID additional (which is merged and released). Could @GaretJax give it a look regarding the code?

**Pull request closed without merging**